src/multicellular_stage/editor/CellBodyPlanEditorComponent.cs

```diff
@@ -477,13 +549,96 @@ public bool ShowCellOptions()
         return true;
     }
 
-    protected CellType CellTypeFromName(string name)
+    public float CalculateSpeed()
     {
-        return Editor.EditedSpecies.CellTypes.First(c => c.TypeName == name);
+        var leader = editedMicrobeCells[0].Data!;
```

Comment:
This should also go into the new cell body plan calculation helpers class as this is so long and I assume auto-evo will also need this info.